### PR TITLE
refactor: replace any with proper types

### PR DESCRIPTION
**Pull request by shroomist**
## Proposed changes

Some progress on Finalise TypeScript Migration #2206 
Addressing little portion of `any` types. Let me know if should proceed in similar fashion.

## Types of changes

- [x] Refactoring (improve a current implementation without adding a new feature or fixing a bug)

## Checklist
- [x] I have read the [CONTRIBUTING](https://docs.ark.io/guidebook/contribution-guidelines/contributing.html) documentation
- [x] Lint and unit tests pass locally with my changes

### CODE REVIEW

File: packages/core-interfaces/src/core-container/container.ts [hunk 1]
```diff
@@ -5,14 +5,18 @@ export interface PluginDescriptor {
     pkg: any;
     defaults?: any;
     extends?: string;
-    register(container: IContainer, options?: any): Promise<any>;
+    register(container: IContainer, options?: PluginRegisterOptions): Promise<any>;
     deregister?(container: IContainer, options?: any): Promise<void>;
 }
 
+export interface PluginRegisterOptions {
```
Comment (faustbrian) [inline packages/core-interfaces/src/core-container/container.ts]: Interfaces should always start with `I` - https://palantir.github.io/tslint/rules/interface-name and `IPluginOptions` might be a better name as the options are for the plugin overall _(used after the registration)_ not just the registration process.

File: packages/core-interfaces/src/core-container/container.ts [hunk 2]
```diff
@@ -5,14 +5,18 @@ export interface PluginDescriptor {
...
+    [key: string]: string | number | object;
```
Comment (faustbrian) [inline packages/core-interfaces/src/core-container/container.ts]: `boolean` and `array` are also possible types for plugin options.

**Comment by faustbrian** [comment]
@shroomist anything more coming in this PR or is it ready?

**Comment by shroomist** [comment]
@faustbrian can be merged

**Pull request merged**